application/common/dos.py

```diff
@@ -102,21 +103,30 @@ def is_valid(self) -> bool:
         return None not in (self.easting, self.northing, self.latitude, self.longitude)
 
 
-def get_matching_dos_services(odscode: str) -> List[DoSService]:
+def get_matching_dos_services(odscode: str, org_type_id: str) -> List[DoSService]:
     """Retrieves DoS Services from DoS database
 
     Args:
         odscode (str): ODScode to match on
+        org_type_id (str): OrganisationType to match on
 
     Returns:
         list[DoSService]: List of DoSService objects with matching first 5
         digits of odscode, taken from DoS database
     """
-
-    logger.info(f"Searching for DoS services with ODSCode that matches first 5 digits of '{odscode}'")
-
-    sql_query = f"SELECT {', '.join(DoSService.db_columns)} FROM services WHERE odscode LIKE %(ODS_5)s"
-    named_args = {"ODS_5": f"{odscode[0:5]}%"}
+    if org_type_id == PHARMACY_ORG_TYPE_ID:
+        logger.info(
+            f"Searching for '{org_type_id}' DoS services with ODSCode that matches first 5 digits of '{odscode}'"
+        )
+        sql_query = f"SELECT {', '.join(DoSService.db_columns)} FROM services WHERE odscode LIKE %(ODS_5)s"
+        named_args = {"ODS_5": f"{odscode[0:5]}%"}
+    elif org_type_id == DENTIST_ORG_TYPE_ID:
+        odscode = get_new_odscode_for_dentist(odscode)
+        logger.info(f"Searching for '{org_type_id}' DoS services with ODSCode that matches '{odscode}'")
+        sql_query = f"SELECT {', '.join(DoSService.db_columns)} FROM services WHERE odscode LIKE %(ODS)s"
+        named_args = {"ODS": f"{odscode}%"}
+    else:
+        logger.warning(f"Found invalid nhsuk org_type_id':{org_type_id}' for a given odscode '{odscode}'")
```

Comment:
What happens if this part runs? It looks like it will log and then cause an exception when it tries to run thew query below.